packages/jupyter-chat/src/selection-watcher.ts

```diff
@@ -207,13 +225,15 @@ function getTextSelection(
     [start, end] = [end, start];
   }
 
+  const language = await languages.getLanguage(editor?.model.mimeType);
```

Comment:
Note `getLanguage` return type is `Promise<IEditorLanguage | null>` so it also returns "not set" in an explicit manner but using `null`.